src/controllers/libraryController.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as chokidar from "chokidar";
+import * as fse from "fs-extra";
+import * as _ from "lodash";
+import { commands, Disposable, ExtensionContext, Uri, window, workspace, WorkspaceFolder } from "vscode";
+import { instrumentOperation } from "vscode-extension-telemetry-wrapper";
+import { Commands } from "../commands";
+import { Jdtls } from "../java/jdtls";
+import { Settings } from "../settings";
+import { Utility } from "../utility";
+import { DataNode } from "../views/dataNode";
+
+export class LibraryController implements Disposable {
+
+    private libraryWatcher: chokidar.FSWatcher | undefined = undefined;
```

Comment:
I don't see anyone is using this watcher. Should remove it?

Reply:
Oh, I forgot to remove this line.

Reply:
I've update the PR. Apart from removing the watcher, I also refactor the controllers to dispose the commands by itself, rather than pushing it to extension context.